File: javascript/String/reverseString.js
function reverse(str) {
    let revd = "";
    for (let letter of str.split('')) {
        revd = letter + revd;
    }
    return revd;
}

function reverseString(str) {
    return str.split('').reverse().join('');
}

function reverseStringRecursion(str) {
    if(str === ''){
        return '';
    } else {
        return reverseStringRecursion(str.substring(1)) + str.charAt(0);
    }
}

function reverseStringLoop(str){
    let result = "";
    for(let i = str.length - 1; i >= 0; i--){
      result += str[i];
    }
    return result;
}

var word = "ocean";
console.log(reverse(word));
console.log(reverseString(word));
console.log(reverseStringRecursion(word));
console.log(reverseStringLoop(word));

File: go/String/reverseString.go
package main

func reverseString(str string) string {
	runed := []rune(str)
	for i, j := 0, len(runed)-1; i < len(runed)/2; i, j = i+1, j-1 {
		runed[i], runed[j] = runed[j], runed[i]
	}
	return string(runed)
}

func main() {
	print(reverseString("reverse me"))
}
